seatunnel-connectors-v2/connector-databend/src/main/java/org/apache/seatunnel/connectors/seatunnel/databend/sink/DatabendSinkAggregatedCommitter.java

```diff
@@ -0,0 +1,296 @@
+
+    private Connection connection;
+    private boolean isCdcMode;
+    // Store catalog table to access schema information
+    private CatalogTable catalogTable;
+
+    // Add a setter for catalogTable
+    public void setCatalogTable(CatalogTable catalogTable) {
+        this.catalogTable = catalogTable;
+    }
+
+    public DatabendSinkAggregatedCommitter(
+            DatabendSinkConfig databendSinkConfig,
+            String database,
+            String table,
+            String rawTableName,
+            String streamName) {
+        this.databendSinkConfig = databendSinkConfig;
+        this.database = database;
+        this.table = table;
+        this.rawTableName = rawTableName;
+        this.streamName = streamName;
+        this.isCdcMode = databendSinkConfig.isCdcMode();
+    }
+
+    @Override
+    public void init() {
+        try {
+            log.info("[Instance {}] Initializing DatabendSinkAggregatedCommitter", instanceId);
+            log.info("[Instance {}] DatabendSinkConfig: {}", instanceId, databendSinkConfig);
+            log.info("[Instance {}] Database: {}", instanceId, database);
+            log.info("[Instance {}] Table: {}", instanceId, table);
+            log.info("[Instance {}] Is CDC mode: {}", instanceId, isCdcMode);
+
+            this.connection = DatabendUtil.createConnection(databendSinkConfig);
+            log.info(
+                    "[Instance {}] Databend connection created successfully: {}",
+                    instanceId,
+                    connection);
+
+            // CDC infrastructure is now initialized in DatabendSink.setJobContext
+            // Just log that we're in CDC mode
+            if (isCdcMode) {
+                log.info("[Instance {}] Running in CDC mode", instanceId);
+            }
+        } catch (SQLException e) {
+            log.error(
+                    "[Instance {}] Failed to initialize DatabendSinkAggregatedCommitter: {}",
+                    instanceId,
+                    e.getMessage(),
+                    e);
+            throw new DatabendConnectorException(
+                    DatabendConnectorErrorCode.CONNECT_FAILED,
+                    "Failed to initialize DatabendSinkAggregatedCommitter: " + e.getMessage(),
+                    e);
+        } catch (Exception e) {
+            log.error(
+                    "[Instance {}] Unexpected error during initialization: {}",
+                    instanceId,
+                    e.getMessage(),
+                    e);
+            throw e;
+        }
+    }
+
+    private String getCurrentTimestamp() {
+        return LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMddHHmmssSSS"));
+    }
+
+    @Override
+    public List<DatabendSinkAggregatedCommitInfo> commit(
+            List<DatabendSinkAggregatedCommitInfo> aggregatedCommitInfos) throws IOException {
+        // Perform final merge operation in CDC mode only when necessary
+        if (isCdcMode && shouldPerformMerge()) {
```

Comment:
```suggestion
        if (isCdcMode) {
```

We should execute merge when commit method be invoked every times. The interval of this method be invoked equals `checkpoint.interval`

Reply:
Perhaps by adding the shouldPerformMerge check, I can reduce some unnecessary merge into operations and alleviate the pressure on the target side?